mlir/include/mlir/Dialect/OpenMP/OpenMPOps.td

```diff
@@ -82,6 +82,15 @@ def ParallelOp : OpenMP_Op<"parallel", [AttrSizedOperandSegments]> {
 
   let parser = [{ return parseParallelOp(parser, result); }];
   let printer = [{ return printParallelOp(p, *this); }];
+  let extraClassDeclaration = [{
```

Comment:
This piece of code is still in review. 
https://reviews.llvm.org/D89340

Reply:
Would be nice if MLIR TableGen could generate these. Did you try this approach with River?